### PR TITLE
Login improvements

### PR DESCRIPTION
**Pull request by joaquimrocha**
Fixes some issues and improves the UX a bit in the login dialog.

### CODE REVIEW

File: frontend/src/lib/k8s/apiProxy.ts [hunk 1]
```diff
@@ -45,8 +49,24 @@ export async function request(path: string, params: RequestParams = {},
...
+
   const url = combinePath(BASE_HTTP_URL, fullPath);
-  const response = await fetch(url, opts);
+  let response: Response = new Response(undefined,
+                                        {status: 502, statusText: 'Unreachable'});
+
+  try {
+    response = await fetch(url, {signal: controller.signal, ...opts});
+  } catch(err) {
+    // AbortController sets the error code as 20
```
Comment (knrt10) [inline frontend/src/lib/k8s/apiProxy.ts:62]:
```suggestion
    // AbortController sets the error code as 20.
```

File: frontend/src/components/authchooser/index.tsx [hunk 2]
```diff
@@ -90,24 +97,37 @@ function AuthChooser(){
...
             if (!useToken) {
               history.replace(from);
             }
 
-            setTestingAuth(false);
-
             // If we reach this point, then we know whether or not we need a token. If we don't,
             // just redirect.
-            if (!cluster.useToken) {
+            if (cluster.useToken === false) {
```
Comment (knrt10) [inline frontend/src/components/authchooser/index.tsx:130]: How about just?

```suggestion
            if (useToken === false) {
```

**Review by knrt10 — CHANGES_REQUESTED** [review]
Some suggestions. Feel free to resolve if it does'nt seems right

File: frontend/src/lib/k8s/apiProxy.ts (hunk 1 above)
Comment (joaquimrocha) [reply frontend/src/lib/k8s/apiProxy.ts:62]: I use periods for comments that are longer than one line. One single line is optional.

File: frontend/src/components/authchooser/index.tsx (hunk 2 above)
Comment (joaquimrocha) [reply frontend/src/components/authchooser/index.tsx:130]: This is deliberate. I want to make sure we're analysing the `cluster`'s property and not the variable. This is usually safer code in this context, as if later we don't assign the `useToken` directly, we still rely on what the cluster's property is.

**Pull request merged**